app/models/manageiq/providers/azure/event_catcher_mixin.rb

```diff
@@ -0,0 +1,38 @@
+module ManageIQ::Providers::Azure::EventCatcherMixin
+  def parse_event_type(event)
+    event_name = event["eventName"]["value"]
+    event_type = ""
+
+    unless event["authorization"].nil? || event["authorization"]["action"].nil?
+      event_type = parse_event_action(event)
+    end
+    "#{event_type}#{event_name}"
+  end
+
+  def parse_event_action(event)
+    # Example autorization action:
+    # Microsoft.Compute/virtualMachines/deallocate/action
+
+    action = event["authorization"]["action"].split("/")
+    "#{action[1]}_#{action[2]}_"
+  end
+
+  def parse_vm_ref(event)
+    # Example resourceId:
+    # /subscriptions/123456789-a1234-12b4-1234-5cd67890312/resourceGroups/
+    # rg_name/providers/Microsoft.Compute/virtualMachines/vm_name
+
+    resource_id = event["resourceId"].downcase.split("/")
+    return nil if resource_id.length < 9
+    join(
```

Comment:
I don't really like this method being separated out...would be cleaner just to have it directly in the method.